design/cluster-api-provider-metal3/node_reuse.md

```diff
@@ -0,0 +1,250 @@
+<!--
+This work is licensed under a Creative Commons Attribution 3.0
+Unported License.
+
+http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# node reuse
+
+## Status
+
+implementable
+
+## Summary
+
+Node reuse during the upgrade/remediation operation.
+
+## Motivation
+
+While performing upgrade operation and potentially remediation, it is desirable
+to be able to reuse the same pool of hosts during re-provisioning (i.e. when
+BareMetalHost is going through deprovisioning -> provisioning). However, there
```

Comment:
What's the effective benefit of reusing the same pool? I think it could be useful to clarify better (ie in term of costs/time saving) such reasons, especially in relationship to the cited scenario (upgrade and/or remediation)

Reply:
So the one case that can benefit from this is upgrade. During the upgrade of Kubeadm Control Plane (KCP) machines, for example we upgrade the K8s version from 1.19 to 1.20, KCP will delete old Machines, thus Metal3Machines will be deleted. As such, BMHs de-provisioning will start. Since this is an upgrade operation, we don't want to lose the date on the external disks that are attached to the hosts. For that we first need to make sure that we disable disk cleaning ([proposal](https://github.com/metal3-io/metal3-docs/blob/master/design/cluster-api-provider-metal3/allow_disabling_node_disk_cleaning.md) and WIP PRs  [BMO](https://github.com/metal3-io/baremetal-operator/pull/784), [CAPM3](https://github.com/metal3-io/cluster-api-provider-metal3/pull/156)) before de-provisioning, and then we try to pick those BMHs that are now in Ready state with external disks left untouched.

I agree that we should mention this explicitly. Will add a note.

Reply:
Agree, _data loss_ can be both costly and time-consuming in case it happens, we can definitely mention it as this will save both time/costs(for retrieving data)

Reply:
@andfasano, we have updated the motivation part now, please take a look and let us know if you think we need more clarification. Thank you.

Reply:
Thanks @fmuyassarov motivations are clear now for me